index.js

```diff
@@ -1,24 +1,46 @@
-let body = document.querySelector("#body")
-let h1 = document.createElement("h1")
-body.append(h1)
-h1.innerText = "Random Star Wars Ship Generator"
-let button = document.createElement("button")
-button.id = "button"
-body.append(button)
-button.innerText = "GET ME A SHIP!"
-let ul = document.createElement("ul")
-body.append(ul)
-let p = document.createElement("p")
-ul.append(p)
-
-button.addEventListener("click", function(evt) {
-    fetch("https:/swapi.dev/api/starships/")
-    .then (res => res.json())
-    .then (function(starship){
-        function randomNumber() {
-            let i = Math.floor(Math.random() * (10))
-            return i
-        }
-        p.innerText = starship.results[randomNumber()].name
-    })
-})
\ No newline at end of file
+function randomShipName(ships) {
+	const index = Math.floor(Math.random() * (ships.length)),
+		ship = ships[index];
+
+	if (ship)
+		return ship.name;
+	else
+		return 'Uh oh! Ship not found :[';
+}
+
+const body = document.querySelector('body'),
+	h1 = document.createElement('h1'),
+	button = document.createElement('button'),
+	p = document.createElement('p');
```

Comment:
it's largely a style preference, but I like declaring a number of variables (`let`, `const`, etc.) at once, so that visually I can see that all of my vars are set in one place.